wgpu/_classes.py

```diff
@@ -2024,23 +2024,31 @@ def messages(self):
 
 
 class GPUQuerySet(GPUObjectBase):
-    """TODO"""
+    """An object to store the results of queries on passes.
 
-    # IDL: undefined destroy();
-    def destroy(self):
-        """Destroy the queryset."""
-        raise NotImplementedError()
+    You can obtain a query set object via :attr:`GPUDevice.create_query_set`.
+    """
+
+    def __init__(self, label, internal, device, type, count):
+        super().__init__(label, internal, device)
+        self._type = type
+        self._count = count
 
     # IDL: readonly attribute GPUQueryType type;
     @property
     def type(self):
         """The type of the queries managed by this queryset."""
-        raise NotImplementedError()
+        return self._type
 
     # IDL: readonly attribute GPUSize32Out count;
     @property
     def count(self):
-        """The type of the queries managed by this queryset."""
+        """The number of the queries managed by this queryset."""
+        return self._count
+
+    # IDL: undefined destroy();
+    def destroy(self):
+        """Destroy this QuerySet."""
         raise NotImplementedError()
```

Comment:
Can't ee implement this?

Reply:
This is the base class. All `destroy()` methods in this module raise NotImplemented, and are implemented in `wgpu_native/_api.py`. I think it makes sense for these here to just call `_destroy()` instead, but since there are multiple such cases, we can leave that for another day imo.

Reply:
To provide some context: almost all classes have a `_destroy()` method that gets automatically called when the object is deleted by the garbage collector (via `__del__`). But the WebGPU spec also gives a few objects an explicit destroy method, which is what we have here.

Reply:
Makes sense!